converters/schema.ts

```diff
@@ -0,0 +1,75 @@
+import { ClarityDataSchema, ObjectType } from "../declarations/clarity";
+import { hashCode } from "./../src/utils";
+
+export class SchemaManager {
+
+  private schemas = {};
+
+  public getSchema(schemaHash: string): ClarityDataSchema {
+    return this.schemas[schemaHash];
+  }
+
+  public addSchema(schema: ClarityDataSchema): boolean {
+    let schemaHash = hashCode(JSON.stringify(schema));
+    let isNew = (typeof this.getSchema(schemaHash) === "undefined");
+    this.schemas[schemaHash] = schema;
+    return isNew;
+  }
+
+  public createSchema(data: any, name?: string): ClarityDataSchema {
+    let schema: ClarityDataSchema = name || null;
+    if (data !== null) {
+      if (typeof data === "object") {
+        schema = this.createNestedSchema(data, name);
+      } else {
+        schema = typeof name === "string" ? name : null;
+      }
+    }
+    return schema;
+  }
+
+  public getSchemaHashcode(schema: ClarityDataSchema) {
+    return hashCode(JSON.stringify(schema));
+  }
+
+  private createNestedSchema(data: object | any[], name?: string): ClarityDataSchema {
```

Comment:
>createNestedSchema [](start = 10, length = 18)

Add some comments which include the examples of schemas that it will produce. It will make it easier for future developers to maintain this code.